crates/sncast/src/starknet_commands/script/init.rs

```diff
@@ -0,0 +1,154 @@
+            Err(err)
+        }
+    }
+}
+
+fn get_script_root_dir_path(script_name: &str) -> Result<Utf8PathBuf> {
+    let current_dir = Utf8PathBuf::from_path_buf(std::env::current_dir()?)
+        .expect("Failed to create Utf8PathBuf for the current directory");
+
+    let scripts_dir = current_dir.join(SCRIPTS_DIR);
+
+    ensure!(
+        !scripts_dir.exists(),
+        "Scripts directory already exists at `{scripts_dir}`"
+    );
+
+    Ok(scripts_dir.join(script_name))
+}
+
+fn init_scarb_project(script_name: &str, script_root_dir: &Utf8PathBuf) -> Result<()> {
+    ScarbCommand::new()
+        .args([
+            "new",
+            "--name",
+            &script_name,
+            "--no-vcs",
+            "--quiet",
+            script_root_dir.as_str(),
+        ])
+        .run()
+        .context("Failed to init Scarb project")?;
+
+    Ok(())
+}
+
+fn add_dependencies(script_root_dir: &Utf8PathBuf) -> Result<()> {
+    add_sncast_std_dependency(script_root_dir)
+        .context("Failed to add sncast_std dependency to Scarb.toml")?;
+    add_starknet_dependency(script_root_dir)
+        .context("Failed to add starknet dependency to Scarb.toml")?;
+
+    Ok(())
+}
+
+fn add_sncast_std_dependency(script_root_dir: &Utf8PathBuf) -> Result<()> {
+    let cast_version = format!("v{}", env!("CARGO_PKG_VERSION"));
+
+    ScarbCommand::new()
+        .current_dir(script_root_dir)
+        .args([
+            "--offline",
+            "add",
+            "sncast_std",
+            "--git",
+            "https://github.com/foundry-rs/starknet-foundry.git",
+            "--tag",
+            &cast_version,
+        ])
+        .run()?;
+
+    Ok(())
+}
+
+fn add_starknet_dependency(script_root_dir: &Utf8PathBuf) -> Result<()> {
+    let scarb_manifest_path = script_root_dir.join("Scarb.toml");
+    let cairo_version =
+        get_cairo_version(&scarb_manifest_path).context("Failed to get cairo version")?;
+    let starknet_dependency = format!("starknet@>={cairo_version}");
+
+    ScarbCommand::new()
+        .current_dir(script_root_dir)
+        .args(["--offline", "add", &starknet_dependency])
+        .run()?;
+
+    Ok(())
+}
+
+fn modify_files_in_src_dir(script_name: &str, script_root_dir: &Utf8PathBuf) -> Result<()> {
+    create_script_main_file(script_name, script_root_dir)
+        .context(format!("Failed to create {script_name}.cairo file"))?;
+    overwrite_lib_file(script_name, script_root_dir).context("Failed to overwrite lib.cairo file")
+}
+
+fn create_script_main_file(script_name: &str, script_root_dir: &Utf8PathBuf) -> Result<()> {
+    let script_main_file_name = format!("{script_name}.cairo");
+    let script_main_file_path = script_root_dir.join("src").join(script_main_file_name);
+
+    fs::write(
+        script_main_file_path,
+        indoc! {r"
+            use sncast_std;
+            use debug::PrintTrait;
+
+            fn main() {
+                'Put your code here!'.print();
```

Comment:
[nitpick]: maybe we could put something more usable here as well? Having a simple `call` to some simple contract displaying "hello world" on testnet would be soo cool, and would actually show at least some of the scripts' functionality